java/server/src/org/openqa/selenium/grid/sessionmap/jdbc/JdbcBackedSessionMap.java

```diff
@@ -0,0 +1,184 @@
+    }
+  }
+
+  @Override
+  public Session get(SessionId id) throws NoSuchSessionException {
+    Require.nonNull("Session ID", id);
+
+    URI uri = null;
+    Capabilities caps = null;
+
+    try {
+      ResultSet sessions = readSessionStatement(id).executeQuery();
+
+      while(sessions.next()) {
+        String sessionIdAndURI = sessions.getString(sessionIdCol);
+        String rawUri = sessionIdAndURI.split(separator)[1];
+        String rawCapabilities = sessions.getString(sessionCapsCol);
+
+        caps = rawCapabilities == null ?
+                            new ImmutableCapabilities() :
+                            JSON.toType(rawCapabilities, Capabilities.class);
+
+        try {
+          uri = new URI(rawUri);
+        }
+        catch (URISyntaxException e) {
+          throw new NoSuchSessionException(String.format("Unable to convert session id (%s) to uri: %s", id, rawUri), e);
+        }
+      }
+
+      sessions.close();
+
+      if (uri == null) {
+        throw new NoSuchSessionException("Unable to find URI for session " + id);
+      }
+
+      return new Session(id, uri, caps);
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void remove(SessionId id) {
+    Require.nonNull("Session ID", id);
+
+    try {
+      getDeleteSqlForSession(id).executeUpdate();
+    } catch (SQLException e) {
+      throw new JdbcException(e.getMessage());
+    }
+  }
+
+  @Override
+  public void close() {
+    try {
+      connection.close();
+    } catch (SQLException e) {
+      LOG.warning("SQL exception while closing JDBC Connection:" + e.getMessage());
+    }
+  }
+
+  private PreparedStatement insertSessionStatement(Session session) throws SQLException {
+    PreparedStatement insertStatement = connection.prepareStatement("insert into " + tableName + " values(?,?)");
+    insertStatement.setString(1, sessionUri(session.getId()) + separator + session.getUri().toString());
+    insertStatement.setString(2, JSON.toJson(session.getCapabilities()));
+
+    return insertStatement;
+  }
+
+  private PreparedStatement readSessionStatement(SessionId sessionId) throws SQLException {
+    PreparedStatement getSessionsStatement = connection.prepareStatement("select * from " + tableName + " where " + sessionIdCol + " like ?");
+    getSessionsStatement.setString(1, sessionUri(sessionId).concat("%"));
+
+    return getSessionsStatement;
+  }
+
+  private PreparedStatement getDeleteSqlForSession(SessionId sessionId) throws SQLException{
+    PreparedStatement deleteSessionStatement = connection.prepareStatement("delete from " + tableName + " where " + sessionIdCol + " like ?");
+    deleteSessionStatement.setString(1, sessionUri(sessionId).concat("%"));
+
+    return deleteSessionStatement;
+
+  }
+
+  private String sessionUri(SessionId sessionId) {
+    Require.nonNull("Session ID", sessionId);
+
+    return "session:" + sessionId.toString() + ":uri";
```

Comment:
The session id should be a UUID (it will be unique, even if it's not formatted correctly)